docs/src/main/asciidoc/elasticsearch-dev-services.adoc

```diff
@@ -0,0 +1,63 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Dev Services for Elasticsearch
+
+include::./attributes.adoc[]
+
+If any Elasticsearch-related extension is present (e.g. `quarkus-elasticsearch-rest-client` or `quarkus-hibernate-search-orm-elasticsearch`),
+Dev Services for Elasticsearch automatically starts an Elasticsearch server in dev mode and when running tests.
+So, you don't have to start a server manually.
+The application is configured automatically.
+
+== Enabling / Disabling Dev Services for Elasticsearch
+
+Dev Services for Elasticsearch is automatically enabled unless:
+
+- `quarkus.elasticsearch.devservices.enabled` is set to `false`
+- the hosts property is configured, depending of the extension used it can be:
+    - `quarkus.elasticsearch.hosts`
+    - `quarkus.hibernate-search-orm.elasticsearch.hosts`
+
+Dev Services for Elasticsearch relies on Docker to start the server.
+If your environment does not support Docker, you will need to start the server manually, or connect to an already running server.
+
+== Shared Elasticsearch
+
+Sometimes you need to share the server between applications.
+Dev Services for Elasticsearch implements a _service discovery_ mechanism for your multiple Quarkus applications running in _dev_ mode to share a single server.
+
+NOTE: Dev Services for Elasticsearch starts the container with the `quarkus-dev-service-elasticsearch` label which is used to identify the container.
+
+If you need multiple (shared) servers, you can configure the `quarkus.elasticsearch.devservices.service-name` attribute and indicate the server name.
+It looks for a container with the same value, or starts a new one if none can be found.
+The default service name is `elasticsearch`.
+
+Sharing is enabled by default in dev mode, but disabled in test mode.
+You can disable the sharing with `quarkus.elasticsearch.devservices.shared=false`.
+
+== Setting the port
+
+By default, Dev Services for Elasticsearch picks a random port and configures the application.
+You can set the port by configuring the `quarkus.elasticsearch.devservices.port` property.
+
+Note that the Elasticsearch hosts property is automatically configured with the chosen port.
+
+== Configuring the image
+
+Dev Services for Elasticsearch only support Elasticsearch images provided by Elastic.
+
+If you need to use a different image than the default one you can configure it via
+[source, properties]
+----
+quarkus.elasticsearch.devservices.image-name=docker.elastic.co/elasticsearch/elasticsearch:7.17.0
+----
+
+== Current limitations
+
+Currently, Elasticsearch Dev Services cannot be used if both Hibernate Search ORM Elasticsearch and an Elasticsearch client extension are used,
+this will result on a build extension so one of the two extension must have Dev Services disabled.
```

Comment:
Yes, here we have two options:
- Moving the de-activation in the user extension
- Implementing the multi stuff: if two extensions wants devservices, gives them and set the two properties

I prefere to implement 2 ;) but maybe on a followup PR

Reply:
I doubt the second solution quites cut it, since most of the time you will want to connect two extensions to the same ES cluster (typically when using the REST client + Hibernate Search in the same app), and sometimes you'll want to use different clusters (though on instinct I'd say it's more rare?). Allowing to differentiate between those cases seems... complicated? Or maybe you'd rely on the "label" thingy that you already expose?

Anyway... Whatever the solution we end up with, at least the documentation should be consistent with the features added in *this* PR :)

Reply:
I two extensions want an Elasticsearch cluster but not the same (which should be rare), one of the two should disabled devservices as for now we didn't support multiple containers. 

So definitly we should do both: moving the de-activation at the consuming extension side and support multiple extensions that request "the same" ES, currently only the default one is supported so it's always this one.

The label thing is a most complex use case when you run two apps that use the same container. Let's keep it this way for now ;)

> Anyway... Whatever the solution we end up with, at least the documentation should be consistent with the features added in this PR :)

I don't know when 2.8 will be cut but if I have time I'll what I just described, otherwise I'll add clarification to the documentation.

Reply:
Great! As 2.8 will not be released soon, I'll plan to adress it before CR1 if I find the time.